data/modifications/pick_advance/main.lua

```diff
@@ -0,0 +1,249 @@
+-- << pick_advance/main.lua
+
+local on_event = wesnoth.require("lua/on_event.lua")
+local T = wesnoth.require("lua/helper.lua").set_wml_tag_metatable {}
+
+wesnoth.wml_actions.set_menu_item {
+	id = "pickadvance",
+	description = "Pick Advance",
```

Comment:
needs _, and due to that the textdomain needs be declared too.

This file uses _ already for something else, but I think it does not clash.

Reply:
`_` is really not the best approach though, it means to mark a variable as unused in Lua... For example, for tools like `luacheck` it is the default convention and you don't get `unused variable` warnings for underscore variables. If you're really, really not interesting in an argument in Lua, you use `_`, just the underscore.

If I would use Lua best practices in Wesnoth, I'd name the function `translate` or `i18n` (or what's the better word from translation engine perspective?), e.g.:

```
local translate = wesnoth.textdomain "wesnoth"
local something = translate "Sword"
local unit = translate("Pikeman")
```

Reply:
The convention for Wesnoth's codebase though is to use an underscore, since that mimics how it's done with WML translations.

Reply:
On the Lua side in can be called anything. Using something else than _ is a matter of adding support in wmlxgettext to also recognize it.

Reply:
@sevu I'd even say wmlxgettext doesn't need to be changed at all, it's a matter of local variable name `_` or `translate` or else. I guess we can close this part of the discussion as offtopic though. I should raise a separate issue for that, if I'd want to change the current Lua guidelines in wesnoth